test/e2e/storage/volume_provisioning.go

```diff
@@ -716,6 +786,37 @@ var _ = utils.SIGDescribe("Dynamic Provisioning", func() {
 			Expect(claim.Status.Phase).To(Equal(v1.ClaimPending))
 		})
 	})
+
+	framework.KubeDescribe("GlusterDynamicProvisioner", func() {
+		It("should create and delete persistent volumes [fast]", func() {
+			By("creating a Gluster DP server Pod")
+			pod := startGlusterDpServerPod(c, ns)
+			serverUrl := "https://" + pod.Status.PodIP + ":8081"
+
+			By("creating a StorageClass")
+			test := storageClassTest{
+				name:         "Gluster Dynamic provisioner test",
+				provisioner:  "kubernetes.io/glusterfs",
+				claimSize:    "1500Mi",
+				expectedSize: "1500Mi",
+				parameters:   map[string]string{"resturl": serverUrl},
+			}
+			class := newStorageClass(test, ns, "gluster")
+			_, err := c.Storage().StorageClasses().Create(class)
+			defer c.Storage().StorageClasses().Delete(class.Name, nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("creating a claim with an alpha dynamic provisioning annotation")
+			claim := newClaim(test, ns, "default")
+			defer func() {
```

Comment:
The claim will attempted to be deleted even if it was not successfully created. Prefer to see the defer Delete after the `Expect` below.

Reply:
Also the _suffix_ arg is not used by the `newClaim` func

Reply:
Moved the `defer()`  after `expect` and also defined `suffix` arg.